lms/static/scripts/frontend_apps/components/Spinner.js

```diff
@@ -1,21 +1,26 @@
-import { createElement } from 'preact';
+import { createElement, Fragment } from 'preact';
 import propTypes from 'prop-types';
 
 import SvgIcon from './SvgIcon';
 import { trustMarkup } from '../utils/trusted';
 /**
  * A spinning loading indicator.
  */
-export default function Spinner({ className }) {
+export default function Spinner({ className, hide }) {
   return (
-    <SvgIcon
-      className={className}
-      src={trustMarkup(require('../../../images/spinner.svg'))}
-      inline={true}
-    />
+    <Fragment>
```

Comment:
The spinner has an option `hide` prop that makes testing a bit easier since the component does not remove itself from the DOM.

Reply:
I'm not super keen on this approach since it means adding a prop like this to each component where you need to wait for it to be removed.

In this branch I've pushed a change which reverts to adding/removing the `Spinner`, but simplifies waiting for the spinner to be shown or hidden by creating a couple of utilities:

```js
function spinnerVisible(wrapper) {
  return waitForElement(wrapper, 'Spinner');
}

function spinnerHidden(wrapper) {
  return waitFor(() => {
    wrapper.update();
    return !wrapper.exists('Spinner');
  });
}
```

The advantage is that this is easier to adapt for any new use cases we have that need to do the same thing.

I think the `waitForElementToBeRemoved` utility that you had implemented previously was also a fine approach here.